### ISSUE TITLE
No light reflection/penumbra shadows

### ISSUE THREAD

**Issue by Rytimiscurisin** [Roblox/future-is-bright#54]
This may not be a bug, but it is certainly a big part of lighting that has been bugging me since I started testing out shadowmapping. Take this image for example: 14:00:00, and outdoor ambient set to [127,127,127]; a typical time of day and brightness. The lights in this room are brighter than usual because they were not made with shadowmapping in mind. 
![image](https://user-images.githubusercontent.com/30421269/28752760-17727e4a-74f5-11e7-9cb6-e5a199c19ecf.png)
Then, change the time of day to 0:00:00, and the outdoor ambient to [21,21,21]; a darker, night time set up. The surface that is directly hit by the light is just as bright, however, the walls around the room are pitch black. 
![image](https://user-images.githubusercontent.com/30421269/28752770-5bc1b9ee-74f5-11e7-8a76-16e08254605e.png)
In a realistic scenario, the light hitting the floor of the building would reflect partially and bounce around the room, lighting up the walls partially, making a penumbra shadow (This is stuff I remember from high school physics courses, I can't remember everything perfectly). 

Another example of this occurs frequently in situations where a light source is enclosed in a lantern, for example: At 0:00:00, and outdoor ambient at [21,21,21], most of the saloon is dark despite the bright lantern in the center of the room, because the metal of the lantern is blocking the light. In the bottom right of the photo, chairs and tables can also be seen leaving a very dark shadow. 
![image](https://user-images.githubusercontent.com/30421269/28752837-b7d16a08-74f6-11e7-8cb2-8c9c257db5be.png)
Again, in a real scenario, the penumbra shadows left by the lantern and props would be much less intense than this. I don't know if this is a fixable issue, but something will need to be changed before shadowmapping is officially added into ROBLOX studio.

**Comment by Kolinchkov** [comment]
Indeed, this has been a bother to me as well. For VoxelCascaded such a feature has been implemented. However, this feature seems to be rather primitive due to the nature of not properly adjusting with brightness levels, as light reflection beyond that remains unaffected. Hopefully said feature will be carried over to ShadowMap in a future demonstration release.

**Comment by Knightshade** [comment]
I've noticed this too with shadow map. I love the shadows it creates but the fact that completely enclosed rooms are as bright as the outdoor ambient setting, it breaks it for me.

**Comment by craftsmashbuild** [comment]
@Knightshade I read that there are plans to improve what you speak of about shadowmap

**Comment by tsbuilderboy** [comment]
I was thinking the same thing. I don't think ROBLOX will add that, though, since not even Unreal Engine has a feature like that. There'd be too many calculations going on, and it would cause major performance issues, especially since ROBLOX usually puts low-end PCs in mind when making updates.

**Comment by Hifiveghost5** [comment]
with the 3rd image with the lantern you can easily fix by selecting the light and then going to the properties and change the shadow cutoff
